codesigndocuitests/codesigngroup.go

```diff
@@ -0,0 +1,424 @@
+		if err != nil {
+			return nil, fmt.Errorf("failed to read input: %s", err)
+		}
+		if !anotherExport {
+			break
+		}
+	}
+
+	return collectedCodeSignGroups, nil
+}
+
+// collectExportSelectableCodeSignGroups returns every possible codesigngroup which can be used to export an ipa file
+func collectExportSelectableCodeSignGroups(testRunner IOSTestRunner, installedCertificates []certificateutil.CertificateInfoModel, installedProfiles []profileutil.ProvisioningProfileInfoModel) []export.SelectableCodeSignGroup {
+	bundleIDEntitlemenstMap := map[string]plistutil.PlistData{}
+	bundleIDEntitlemenstMap = testRunner.BundleIDEntitlementsMap()
+
+	bundleIDs := []string{}
+	for bundleID := range bundleIDEntitlemenstMap {
+		bundleIDs = append(bundleIDs, bundleID)
+	}
+	codeSignGroups := export.CreateSelectableCodeSignGroups(installedCertificates, installedProfiles, bundleIDs)
+
+	log.Debugf("Codesign Groups:")
+	for _, group := range codeSignGroups {
+		log.Debugf(group.String())
+	}
+
+	if len(codeSignGroups) == 0 {
+		return []export.SelectableCodeSignGroup{}
+	}
+
+	codeSignGroups = export.FilterSelectableCodeSignGroups(codeSignGroups,
+		export.CreateEntitlementsSelectableCodeSignGroupFilter(bundleIDEntitlemenstMap),
+	)
+
+	fmt.Println()
+	// Handle if UITest target used NON xcode managed profile
+	if len(codeSignGroups) > 0 && !testRunner.IsXcodeManaged() {
+		log.Warnf("The UITest target (%s) was signed with NON xcode managed profile,", path.Base(testRunner.Path))
+		log.Warnf("only NOT xcode managed profiles are allowed to sign the UITest target.")
+		log.Warnf("Removing xcode managed CodeSignInfo groups")
+
+		codeSignGroups = export.FilterSelectableCodeSignGroups(codeSignGroups,
+			export.CreateNotXcodeManagedSelectableCodeSignGroupFilter(),
+		)
+	}
+
+	// Handle if UITest target used NON xcode managed profile
+	if len(codeSignGroups) > 0 && testRunner.IsXcodeManaged() {
+		log.Warnf("The UITest target (%s) was signed with xcode managed profile,", path.Base(testRunner.Path))
+		log.Warnf("only xcode managed profiles are allowed to sign the UITest target.")
+		log.Warnf("Removing NON xcode managed CodeSignInfo groups")
+
+		codeSignGroups = export.FilterSelectableCodeSignGroups(codeSignGroups,
+			export.CreateXcodeManagedSelectableCodeSignGroupFilter(),
+		)
+	}
+
+	log.Debugf("\n")
+	log.Debugf("Filtered Codesign Groups:")
+	for _, group := range codeSignGroups {
+		log.Debugf(group.String())
+	}
+
+	return codeSignGroups
+}
+
+// hasCertificateForDistType returns true if the provided certificate list has certificate for the selected cert type.
+// If isDistCert == true it will search for Distribution Certificates. If it's == false it will search for Developmenttion Certificates.
+// If the team doesn't have any certificate for the selected cert type, it will return false.
+func hasCertificateForDistType(exportMethod string, certificates []certificateutil.CertificateInfoModel) bool {
+	switch exportMethod {
+	case "development":
```

Comment:
if - else would be better as we have two cases only here

Reply:
👍